src/sas/qtgui/Perspectives/Shape2SAS/Constraints.py

```diff
@@ -0,0 +1,352 @@
+            raise ModuleNotFoundError(f"No module named {node.module}")
+
+        imports = []
+        module = importlib.import_module(node.module)
+
+        for alias in node.names:
+            #check if library has the attribute
+            if not hasattr(module, f"{alias.name}"):
+                self.logException(f"AttributeError: module {node.module} has no attribute {alias.name}")
+                raise AttributeError(f"module {node.module} has no attribute {alias.name}")
+            if alias.asname:
+                imports.append(f"{alias.name} as {alias.asname}")
+            else:
+                imports.append(f"{alias.name}")
+        
+        return [f"from {node.module} import {', '.join(imports)}"]
+
+
+    def isImportStatement(self, node: ast.Import) -> list[str]:
+        """Return list of Import statements"""
+
+        imports = []
+        for alias in node.names:
+            #check if library exists
+            if not importlib.util.find_spec(alias.name):
+                self.logException(f"ModuleNotFoundError: No module named {alias.name}")
+                raise ModuleNotFoundError(f"No module named {alias.name}")
+            #get name and asname
+            if alias.asname:
+                imports.append(f"{alias.name} as {alias.asname}")
+            else:
+                imports.append(f"{alias.name}")
+
+        return [f"import {', '.join(imports)}"]
+    
+
+    def getImportStatements(self, text: str) -> list[str]:
+        """return all import statements that were 
+        written in the text editor"""
+
+        importStatements = []
+
+        try:
+            tree = ast.parse(text)
+            #look for import statements
+            for node in ast.walk(tree):
+                #check statement type
+                if isinstance(node, ast.ImportFrom):
+                    importStatements.extend(self.isImportFromStatement(node))
+
+                elif isinstance(node, ast.Import):
+                    importStatements.extend(self.isImportStatement(node))
+            
+            return importStatements
+        
+        except SyntaxError as e:
+            error_line = text.splitlines()[e.lineno - 1]
+            self.logException(f"Syntax error: {e.msg} at line {e.lineno}: {error_line}")
+            raise SyntaxError(f"Syntax error: {e.msg} at line {e.lineno}: {error_line}")
+            return
+
+    
+    def logException(self, exception):
+        htmlise = "<br>".join(exception.split("\n"))
+        self.textEdit_2.append(f'<span style="color:Red;"><b>{htmlise}</b></span>')
+
+    def logWarning(self, message):
+        htmlise = "<br>".join(message.split("\n"))
+        self.textEdit_2.append(f'<span style="color:Orange;"><b>Warning: {htmlise}</b></span>')
+
+    def logInfo(self, message):
+        htmlise = "<br>".join(message.split("\n"))
+        self.textEdit_2.append(f'<span style="color:Black;"><b>Info: </b>{htmlise}</span>')
```

Comment:
To tie into the SasView log, simply use the logging package. Extra logs will only cause more confusion.

```
logger = logging.getLogger(__name__)

...

    logger.<info|warn|error>(message)
```